### PR TITLE
python3 support and make it as a vim plugin

### PR DESCRIPTION
**Pull request by lisongmin**
python2/python3 is all supported with this pr,
and a vim plugin is added so that we can quick install and less configuration.

### CODE REVIEW

**Comment by jszakmeister** [comment]
Thanks, but one issue right off the bat is that I still use this on systems that use Python 2.6, so I'm not willing to lose that support just yet. :-(

File: plugin/tagbar-markdown.vim [hunk 1]
```diff
@@ -0,0 +1,57 @@
+
+if exists("tagbar_markdown")
+    finish
+endif
+let tagbar_markdown = 1
+
+
+if (!exists('g:markdown2ctags'))
+    let g:markdown2ctags = '~/.vim/bundle/markdown2ctags/markdown2ctags.py'
```
Comment (jszakmeister) [inline plugin/tagbar-markdown.vim]: I think there's probably a better way to do this without hard-coding the path quite like this.

Comment (jszakmeister) [inline plugin/tagbar-markdown.vim]: Oh, and while it's longer, I prefer to prefix my global variables more to help avoid collisions... so `g:tagbar_markdown_markdown2ctags` would be more appropriate.

**Comment by jszakmeister** [comment]
So, I didn't write a Vim plugin thinking that I didn't want to confine people to Vim (there's no reason other editors couldn't use this).  But I guess I can see the value in having it too.  I need to think more about it though.

**Comment by lisongmin** [comment]
It should running on python2.6 as well. I test success under sles11SP1 with python2.6.

File: markdown2ctags.py [hunk 2]
```diff
@@ -184,29 +186,29 @@ def genTagsFile(output, tags, sort):
...
 
     if options.tagfile == '-':
         output = sys.stdout
     else:
-        output = open(options.tagfile, 'wb')
+        output = open(options.tagfile, 'w')
 
     for filename in args:
-        f = open(filename, 'rb')
+        f = open(filename, 'r')
```
Comment (jszakmeister) [inline markdown2ctags.py]: How does this work out with files with other encodings?

**Comment by jszakmeister** [comment]
> It should running on python2.6 as well. I test success under sles11SP1 with python2.6.

Hrmph, you're right.  Okay.  I'll try and take a closer look at it this weekend.

File: markdown2ctags.py (hunk 2 above)
Comment (lisongmin) [inline markdown2ctags.py]: Work well under python2, but not python3 in this case. 

```
 ✘ lsm@lsm-pc  /tmp  ~/.vim/bundle/markdown2ctags/markdown2ctags.py -f - a.md
Traceback (most recent call last):
  File "/home/lsm/.vim/bundle/markdown2ctags/markdown2ctags.py", line 223, in <module>
    main()
  File "/home/lsm/.vim/bundle/markdown2ctags/markdown2ctags.py", line 212, in main
    lines = f.read().splitlines()
  File "/usr/lib/python3.5/codecs.py", line 321, in decode
    (result, consumed) = self._buffer_decode(data, self.errors, final)
UnicodeDecodeError: 'utf-8' codec can't decode byte 0xc9 in position 43: invalid continuation byte
```

I will debug it later ...

**Comment by lisongmin** [comment]
There is a bug if filename and content use different encodings. I have no idea how to solve it.

**Comment by jszakmeister** [comment]
I'm just not keen on this approach, and it could be useful for other editors than just Vim.  Thank you and sorry it's been languishing here so long.

**Pull request closed without merging**